checkBoard.c

```diff
@@ -0,0 +1,44 @@
+#include <stdio.h>
+
+int checkBoard(char board[],char marker) {      	       
+	int i,res;
+//   0   1   2
+//   3   4   5
+//   6   7   8
+	if (marker=='X'){
+		res=1;
+	}else{
+		res=-1;
+	}
+	for (i=0; i<=8; i++) {
+        if (board[i]==marker && board[i+1]==marker && board[i+2]==marker ) {   
+			return res;
+		} 
+		i+2;
+	}
+
+	for (i=6; i<8; i++) {
+      	if (board[i]==marker && board[i-3]==marker && board[i-6]==marker ) {    
+				return res;
+		}
+	}
+
+	if (board[0]==marker && board[4]==marker && board[8]==marker ){
+		return res; 
+	}
+   	if (board[6]==marker && board[4]==marker && board[2]==marker ){
+		return res;
+	}
+	
+	int count=0;
```

Comment:
It would be nice to have a comment above each of the logical section with description what is does.